File: php/features.php
<?php
declare(strict_types=1);

// OktiasBakery SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class OktiasBakeryFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new OktiasBakeryBaseFeature();
            case "test":
                return new OktiasBakeryTestFeature();
            default:
                return new OktiasBakeryBaseFeature();
        }
    }
}

File: py/features.py
# OktiasBakery SDK feature factory

from feature.base_feature import OktiasBakeryBaseFeature
from feature.test_feature import OktiasBakeryTestFeature


def _make_feature(name):
    features = {
        "base": lambda: OktiasBakeryBaseFeature(),
        "test": lambda: OktiasBakeryTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
